release.sh

```diff
@@ -41,17 +55,36 @@ if [ -n "$(git status --porcelain)" ]; then
     git commit --signoff -m "Update version to $VERSION"
 fi
 
-echo Linting, building and testing:
+echo "Linting, building and testing:"
 npm run lint
 npm run build
 npm run test
 
-echo Creating Tag...
+echo "Creating Tag ..."
 git tag -a -m "$VERSION_PREFIXED" "$VERSION_PREFIXED"
 
-echo Pushing version and tag to GitHub repository:
-git push
-git push "$(git config --get remote.origin.url)" "$VERSION_PREFIXED"
+if [[ $DRY_RUN ]]; then
+  echo "Pushing version and tag to GitHub repository (dry-run):"
+  git push --set-upstream origin "$RELEASE_BRANCH" --dry-run
+  git push "$(git config --get remote.origin.url)" "$VERSION_PREFIXED" --dry-run
+
+  echo "Publish to npmjs (dry-run):"
+  npm publish --access public --registry https://registry.npmjs.org/ --dry-run
+
+  echo "Deleting temporary release-branch (dry-run):"
+  git checkout main
+  git branch -D "$RELEASE_BRANCH"
+
+else
+  echo "Pushing version and tag to GitHub repository:"
+  git push --set-upstream origin "$RELEASE_BRANCH"
+  git push "$(git config --get remote.origin.url)" "$VERSION_PREFIXED"
+
+  echo "Publish to npmjs:"
+  npm publish --access public --registry https://registry.npmjs.org/
 
-echo Publish to npmjs:
-npm publish --access public --registry https://registry.npmjs.org/
+  echo "Deleting temporary release-branch:"
+  git checkout main
+  git push origin --delete "$RELEASE_BRANCH"
```

Comment:
Don't we want to keep the remote release branch and merge it? This looks to me as if we would not have our version tag on main at the moment after a release.

Reply:
Yes, we should merge the release branch to main after we committed the updated package.json
I guess we need to wait for the github checks to succeed or our github rules won't allow us to merge ... so somehow sleep a while, try to merge, if checks are not ready, sleep more, try again ...?

Reply:
Ah ok, I wasn't aware we wanted to merge this back after the release. Is it possible to merge onto main without having a pull request and review, since we have _main_ as a protected branch?

Reply:
good question, I don't now, maybe we will need to remove the requirement for reviews to be able to do this ... :(

Reply:
We decided to split the release script in two parts. The first script will prepare everything and create a PR. After manually merging the PR the second script will do the actual publishing to npmjs.